src/Compilers/CSharp/Portable/Lowering/MethodToClassRewriter.cs

```diff
@@ -389,7 +389,7 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
                 NeedsProxy(leftLocal.LocalSymbol))
             {
                 Debug.Assert(!proxies.ContainsKey(leftLocal.LocalSymbol));
-                Debug.Assert(!IsStackAlloc(originalRight));
+                Debug.Assert(originalRight.Kind != BoundKind.ConvertedStackAllocExpression);
```

Comment:
I don't understand why we have asserts right before throwing an exception. It will always throw...

Reply:
It should always throw (unreachable code).
However, my understanding is that this should warn the dev debugging this to the fact that `stackalloc` should never be here (and the fact that we are thworing shouldn't hide that).

---
In reply to: [136684142](https://github.com/dotnet/roslyn/pull/21746#discussion_r136684142) [](ancestors = 136684142)